File: app/src/main/java/com/taskstracker/Model/DataBase/TasksLockingManager.java
package com.taskstracker.Model.DataBase;

import com.taskstracker.Model.DataModels.Task;

import java.util.List;

public class TasksLockingManager {
    public static boolean  lockTasks(List<Task> tasksList, Task updatedTask, int newStatus){
        return false;
    }
}

File: app/src/main/java/com/taskstracker/Model/DataBase/TasksLockingManager.kt
package com.taskstracker.Model.DataBase

import com.taskstracker.Model.DataModels.Task

class TasksLockingManager() {
    fun lockTasks(tasks: List<Task>, updatedPos: Int, newStatus: Int) {
        if(newStatus==Task.OPEN){
            tasks.stream().forEach { t ->  t.setIsLocked(false)}
        }
        else{
            tasks.stream().forEach { t ->  t.setIsLocked(true)}
            tasks[updatedPos].setIsLocked(false)
        }
    }
}